pdns/ws-auth.cc

```diff
@@ -2270,6 +2270,14 @@ static void apiServerCacheFlush(HttpRequest* req, HttpResponse* resp) {
 
   DNSName canon = apiNameToDNSName(req->getvars["domain"]);
 
+  SOAData sd;
+  UeberBackend B;
+  if (g_zoneCache.isEnabled() && B.getSOAUncached(canon, sd)) {
+    // zone exists (uncached), add/update it in the zone cache.
+    // Handle this first, to avoid concurrent queries re-populating the other caches.
+    g_zoneCache.add(canon, sd.domain_id);
+  }
+
   uint64_t count = purgeAuthCachesExact(canon);
```

Comment:
```suggestion
  uint64_t count = purgeAuthCaches(canon.toString() + "$");
```
Bonus fix unrelated to the pull ;)

Reply:
added!